lms/djangoapps/courseware/views.py

```diff
@@ -733,7 +733,7 @@ def course_info(request, course_id):
         return render_to_response('courseware/info.html', context)
 
 
-def get_last_accessed_courseware(course, request):
+def get_last_accessed_courseware(course, request, staff_access):
```

Comment:
Instead passing `staff_access` here, you should pass user object. 

Means, don't call `setup_masquerade` twice.